src/bucket/BucketManager.cpp

```diff
@@ -715,6 +716,8 @@ BucketManager::getBucketListReferencedBuckets() const
                 CLOG_TRACE(Bucket, "{} referenced by bucket list",
                            binToHex(*rit.first));
             }
+            // TODO: fix race on FutureBucket (during apply), BucketLevel
```

Comment:
This was fixed by locking mLedgerStateMutex during forgetUnreferencedBuckets, and removing LedgerManager calls from that function to avoid deadlocks. Removed the TODO.